src/background/stats.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import AnonymousCommunication from '@whotracksme/webextension-packages/packages/anonymous-communication';
+import Storage from '@whotracksme/webextension-packages/packages/anonymous-communication/storage';
+import {
+  getTimeAsYYYYMMDDHH,
+  getTimeAsYYYYMMDD,
+  getTrustedUtcTime,
+} from '@whotracksme/webextension-packages/packages/anonymous-communication/timestamps';
+
+const HPN_URL = 'https://collector-hpn.ghostery.net';
+const CONFIG_URL = 'https://api.ghostery.net/api/v1/config';
+
+async function stats() {
+  const storage = new Storage('hpn-public-keys');
+  await storage.init();
+
+  const location = await fetch(CONFIG_URL)
+    .then((res) => res.json())
+    .then((res) => res.location);
+
+  const communication = new AnonymousCommunication({
+    config: {
+      COLLECTOR_DIRECT_URL: HPN_URL,
+      COLLECTOR_PROXY_URL: HPN_URL,
+    },
+    storage,
+  });
+
+  function send() {
+    const now = getTrustedUtcTime();
+    const ts = getTimeAsYYYYMMDD(now);
+    const t = getTimeAsYYYYMMDDHH(now);
+
+    communication.send({
+      action: 'wtm.alive',
+      ver: 1,
+      channel: 'safari',
+      ts,
+      payload: { t, ctry: location },
+    });
+  }
+
+  send();
+  setInterval(send, 1000 * 60 * 60 /* 1 hour */);
+}
+
+stats();
```

Comment:
Would recommend to always handle errors. Mostly for better logs, but also as future JavaScript engines might no longer allow unhandled promises. It is harder to find all these spots later.